File: sdks/csharp/rest/src/Brume.Rest/Projects/Requests/UpdateProjectTierRequest.cs
using Brume.Rest.Core;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

[Serializable]
public record UpdateProjectTierRequest
{
    /// <summary>
    /// One of `"free" | "starter" | "pro" | "business"`.
    /// </summary>
    [JsonPropertyName("tier")]
    public required string Tier { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/rest/src/Projects/Requests/UpdateProjectTierRequest.php
<?php

namespace Brume\Rest\Projects\Requests;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;

class UpdateProjectTierRequest extends JsonSerializableType
{
    /**
     * @var string $tier One of `"free" | "starter" | "pro" | "business"`.
     */
    #[JsonProperty('tier')]
    public string $tier;

    /**
     * @param array{
     *   tier: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->tier = $values['tier'];
    }
}
